bionic/executor.py

```diff
@@ -19,14 +20,21 @@
 _executor = None
 
 
-def get_reusable_executor():
+def get_reusable_executor(worker_count):
     global _executor
 
-    # TODO: For now, executor is always the same. But once the worker
-    # count is configurable, we should try and update the loky executor
-    # and log a warn if the executor changed.
     if _executor is None:
-        _executor = Executor()
+        _executor = Executor(worker_count)
+    else:
+        process_pool_changed = _executor.update_process_pool(worker_count)
+        if process_pool_changed:
+            workers = worker_count if worker_count is not None else "the default value"
+            warning = f"""
```

Comment:
WDYT about something like
```
The number of workers in the global Loky process pool has changed from {old} to {new}.
Because this pool is global, all parallel jobs will now run with the new number of workers,
even in Bionic flows that were configured with the old number.
```
I know this requires tracking a little more state, but I think it will be easier to understand, especially considering that this message will pop up unexpectedly and the user probably won't know what a ProcessPoolExecutor is.

Reply:
Done. I am not tracking the actual worker count, but the value of the `worker_count` sent to this method. The difference between the two is, for the default value of `None`, the warning will print `the default value` instead of the worker count in the `loky` executor. I'm doing this because the `worker_count` in `loky` executor is private.

Reply:
Actually, loky isn't changing the executor when worker count changes. It just resizes the executor. I'm going to change this logic.

Reply:
Fixed